lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -781,6 +781,27 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
+  defmodule User do
+    defstruct [:name]
+  end
+
+  test ":__struct__ is ignored when using structs" do
+    assert capture_err(fn ->
+      Code.eval_string """
+      assert %Kernel.WarningTest.User{__struct__: Ignored, name: "joe"} ==
+             %Kernel.WarningTest.User{name: "joe"}
+      """, _bindings = [], __ENV__
+    end) =~ "key :__struct__ is ignored when using structs"
+
+    assert capture_err(fn ->
+      Code.eval_string """
+      user = %Kernel.WarningTest.User{name: "meg"}
+      assert %Kernel.WarningTest.User{user | __struct__: Ignored, name: "joe"} ==
+             %Kernel.WarningTest.User{__struct__: Kernel.WarningTest.User, name: "joe"}
+      """, _bindings = [], __ENV__
+    end) =~ "key :__struct__ is ignored when using structs"
```

Comment:
We don't use the `_underscored_var = arg` style in Elixir so let's avoid it in here and above please.